pkg/option/resolver/resolver_test.go

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package resolver
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestWriteConfigurations(t *testing.T) {
+	dir := t.TempDir()
+
+	out := map[string]string{
+		"A": "a",
+		"B": "b",
+	}
+
+	err := WriteConfigurations(context.Background(), dir, out)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for k, v := range out {
+		actual, err := os.ReadFile(filepath.Join(dir, k))
+		if err != nil {
+			t.Fatal(err)
+		}
+		if string(actual) != v {
+			t.Fatalf("Unexpected value, wanted %s got %s", v, actual)
+		}
+	}
+}
```

Comment:
Do you think it makes sense to have some golden tests or similar for evaluating that this package does the right thing?

Reply:
very good point. I'll look in to that.

Reply:
This will take a bit of time to write up, since it needs its own suite. Do you mind if it is in a follow-up PR?

I absolutely think full testing of this is critical, and I won't let it slide :-).

Reply:
I'll trust you, sure. :innocent: https://github.com/cilium/cilium/issues/22507